src/looprestoration.rs

```diff
@@ -23,3 +27,188 @@ pub union LooprestorationParams {
     pub filter: Align16<[[int16_t; 8]; 2]>,
     pub sgr: LooprestorationParams_sgr,
 }
+
+macro_rules! looprestoration_filter_fn_enum {
+    (
+        pub enum $enum_name:ident;
+
+        {
+            $(
+                $rust_var:ident => { $rust_fn_16bpc:ident, $rust_fn_8bpc:ident },
+            )*
+        }
+
+        $(
+            #[$arch:meta]
+            {
+                $(
+                    $var_name:ident => { $fn_16bpc:ident, $fn_8bpc:ident },
+                )*
+            }
+        )*
+    ) => {
+        #[derive(Debug, Clone, Copy)]
+        #[repr(u8)]
+        pub enum $enum_name {
+            $(
+                $rust_var,
+            )*
+
+            $($(
+                #[cfg(all(feature = "asm", $arch))]
+                $var_name,
+            )*)*
+        }
+
+        impl $enum_name {
+            #[cfg(feature = "bitdepth_16")]
+            #[inline]
+            pub fn call_16bpc(
```

Comment:
These need to unify bitdepths into one generic signature, because the callers will eventually be deduplicate into generic functions.  I used an extension `trait` of `BitDepth` to accomplish this.